project/index.mjs

```diff
@@ -267,107 +261,118 @@ async function createNewProject(req, res) {
-        `TPEN3 project "${req.params.id}" does not exist.`
+        error.status ?? error.code ?? 500,
+        error.message ?? "Unknown server error"
       )
     }
-  } catch (err) {
-    utils.respondWithError(
-      res,
-      500,
-      "The TPEN3 server encountered an internal error."
-    )
-  }
-})
+  })
+  .all((req, res) => {
+    respondWithError(res, 405, "Improper request method. Use POST instead")
+  })
 
-router.route("/import").post(auth0Middleware(), async (req, res) => {
-   let {createFrom} = req.query
-  createFrom = createFrom?.toLowerCase()
+router
+  .route("/import")
+  .post(auth0Middleware(), async (req, res) => {
+    let {createFrom} = req.query
+    let user = req.user
+    createFrom = createFrom?.toLowerCase()
 
-  if (!createFrom)
-    return res
-      .status(400)
-      .json({
+    if (!createFrom)
+      return res.status(400).json({
         message:
           "Query string 'createFrom' is required, specify manifest source as 'URL' or 'DOC' "
       })
 
-  if (createFrom === "url") {
-    const manifestURL = req?.body?.url
-    
-    let checkURL = await validateURL(manifestURL)
-    
-    if (!checkURL.valid)  return res
-        .status(checkURL.status)
-        .json({message: checkURL.message}) 
+    if (createFrom === "url") {
+      const manifestURL = req?.body?.url
 
-    // return res.json(validation)
+      let checkURL = await validateURL(manifestURL)
 
-    // if (!manifestURL)
-    //   return res
-    //     .status(400)
-    //     .json({message: "Manifest URL is required for import"})
+      if (!checkURL.valid)
+        return res.status(checkURL.status).json({message: checkURL.message})
 
-    try {
-      const result = await ImportProject.fromManifestURL(manifestURL)
-      res.status(201).json(result)
-    } catch (error) {
-      res
-        .status(error.status??500)
-        .json({status: error.status ?? 500, message: error.message})
-    }
-  } else {
-    res
-      .status(400)
-      .json({
+      try {
+        const result = await ProjectFactory.fromManifestURL(manifestURL, user?.agent)
+        res.status(201).json(result)
+      } catch (error) {
+        res
+          .status(error.status ?? 500)
+          .json({status: error.status ?? 500, message: error.message})
+      }
+    } else {
+      res.status(400).json({
         message: `Import from ${createFrom} is not available. Create from URL instead`
       })
-  }
-})
+    }
+  })
+  .all((req, res) => {
+    respondWithError(res, 405, "Improper request method. Use POST instead")
+  })
+
+router
+  .route("/:id")
+  .get(auth0Middleware(), async (req, res) => {
+    let id = req.params.id
+    if (!id) {
+      return respondWithError(res, 400, "No TPEN3 ID provided")
+    } else if (!validateID(id)) {
+      return respondWithError(
+        res,
+        400,
+        "The TPEN3 project ID provided is invalid"
+      )
+    }
+
+    const projectObj = new Project(id)
+    projectObj
+      .getById(id)
```

Comment:
This is how it works now, but it feels goofy. I think `new Project(id)` should just get that Project and return it to you. Just a comment, here, though, not a change request.